frontend/jupyter/src/app/pages/index/index-default/index-default.component.ts

```diff
@@ -406,20 +409,79 @@ export class IndexDefaultComponent implements OnInit, OnDestroy {
     return STATUS_TYPE.TERMINATING;
   }
 
+  // Defines the status of the "Open Viewer" button
+  public parseOpenPVCViewerActionStatus(pvc: PVCProcessedObject): STATUS_TYPE {
+    // PVC is UNAVAILABLE but only because its waiting for a consumer
+    // This shouldn't stop a viewer from being the first consumer
+    const pvcWaitingForConsumer =
+      pvc.status.phase === STATUS_TYPE.UNAVAILABLE &&
+      pvc.status.state === 'WaitForFirstConsumer';
+
+    if (pvc.status.phase !== STATUS_TYPE.READY && !pvcWaitingForConsumer) {
+      return STATUS_TYPE.UNAVAILABLE;
+    }
+
+    // Popup is waiting for the viewer to become ready
+    if (this.pvcsWaitingViewer.has(pvc.name)) {
+      // Open the viewer window if it's ready
+      if (pvc.viewer.status === STATUS_TYPE.READY) {
+        this.pvcsWaitingViewer.delete(pvc.name);
+        this.openViewerWindow(pvc);
+      }
+      // Show a spinner as we're waiting to the viewer to become ready
+      if (
+        [STATUS_TYPE.UNINITIALIZED, STATUS_TYPE.WAITING].includes(
+          pvc.viewer.status,
+        )
+      ) {
+        return STATUS_TYPE.WAITING;
+      }
+    }
+
+    return pvc.viewer.status;
+  }
+
+  // Defines the status of the "Close Viewer" button
+  public parseClosePVCViewerActionStatus(pvc: PVCProcessedObject) {
+    // Users may always close an existing, non-terminating viewer
+    switch (pvc.viewer.status) {
+      case STATUS_TYPE.UNINITIALIZED:
+        return STATUS_TYPE.UNAVAILABLE;
+      case STATUS_TYPE.TERMINATING:
+        return STATUS_TYPE.WAITING;
+      default:
+        return STATUS_TYPE.READY;
+    }
+  }
+
+  public openViewerWindow(pvc: PVCProcessedObject) {
+    const url = this.env.viewerUrl + pvc.viewer.url;
+
+    window.open(url, `${pvc.name}: Volumes Viewer`, 'height=600,width=800');
+  }
+
   public reactVolumeToAction(a: ActionEvent) {
     switch (a.action) {
       case 'delete':
         this.deleteVolumeClicked(a.data);
         break;
+      case 'open-pvcviewer':
+        this.openPVCViewerClicked(a.data);
+        break;
+      case 'close-pvcviewer':
+        this.closePVCViewerClicked(a.data);
+        break;
       case 'name:link':
         if (a.data.status.phase === STATUS_TYPE.TERMINATING) {
           a.event.stopPropagation();
           a.event.preventDefault();
-          this.snackBar.open(
-            'PVC is unavailable now.',
-            SnackType.Warning,
-            3000,
-          );
+          const config: SnackBarConfig = {
+            data: {
+              msg: 'PVC is unavailable now.',
+              snackType: SnackType.Warning,
+            },
+          };
+          this.snackBar.open(config);
```

Comment:
How come this isn't localized while line 500 is? 
I kinda assume localization is in another ticket but if they are both snackbars I would think they behave the same way

Reply:
I dont know. It was like that before. This just changes how the snackbars are generated.

I already made note in the translation ticket to review every snackback message.